src/main/scala/org/apache/spark/sql/pulsar/PulsarProvider.scala

```diff
@@ -366,6 +371,10 @@ private[pulsar] object PulsarProvider extends Logging {
     parameters(ServiceUrlOptionKey)
   }
 
+  private def getAdminUrl(parameters: Map[String, String]): String = {
+    parameters(AdminUrlOptionKey)
```

Comment:
We need to check whether admin url is set because we should allow user to skip this option when read limit is not set.